File: MiPrimeraAplicacion/app/src/main/java/com/micky/miprimeraaplicacion/CalculadoraActivity.java
package com.micky.miprimeraaplicacion;

import androidx.appcompat.app.AppCompatActivity;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.micky.miprimeraaplicacion.backend.Operaciones;

public class CalculadoraActivity extends AppCompatActivity {

    EditText editTextA, editTextB,editTextC;
    Button buttonOperaciones, buttonResta,buttonMultiplicacion,buttonDivision,button1,button2,button3,
            button4,button5,button6,button7,button8,button9,button0,buttonS,buttonR,buttonM,buttonD,buttonC,buttonSalirCalculadora;
    @SuppressLint("MissingInflatedId")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_calculadora);

        editTextA = findViewById(R.id.editTextA);
        editTextB = findViewById(R.id.editTextB);
        editTextC = findViewById(R.id.editTextC);
        buttonOperaciones = findViewById(R.id.buttonOperaciones);
        buttonResta = findViewById(R.id.buttonResta);
        buttonMultiplicacion = findViewById(R.id.buttonMultiplicacion);
        buttonDivision = findViewById(R.id.buttonDivision);
        button1=findViewById(R.id.button1);
        button2=findViewById(R.id.button2);
        button3=findViewById(R.id.button3);
        button4=findViewById(R.id.button4);
        button5=findViewById(R.id.button5);
        button6=findViewById(R.id.button6);
        button7=findViewById(R.id.button7);
        button8=findViewById(R.id.button8);
        button9=findViewById(R.id.button9);
        button0=findViewById(R.id.button0);
        buttonS=findViewById(R.id.buttonS);
        buttonR=findViewById(R.id.buttonR);
        buttonM=findViewById(R.id.buttonM);
        buttonD=findViewById(R.id.buttonD);
        buttonC=findViewById(R.id.buttonC);
        buttonSalirCalculadora=findViewById(R.id.buttonSalirCalculadora);

        Button buttonIgual = findViewById(R.id.buttonIgual);

        buttonOperaciones.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick (View v){
                int x = Integer.parseInt(editTextA.getText().toString());
                int y = Integer.parseInt(editTextB.getText().toString());
                Operaciones operaciones = new Operaciones(x, y);
                Toast.makeText(CalculadoraActivity.this, "La suma es :"+operaciones.suma(), Toast.LENGTH_SHORT).show();
            }
        });

        buttonResta.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick (View v){
                int x = Integer.parseInt(editTextA.getText().toString());
                int y = Integer.parseInt(editTextB.getText().toString());
                Operaciones operaciones = new Operaciones(x, y);
                Toast.makeText(CalculadoraActivity.this, "La resta es :"+operaciones.resta(), Toast.LENGTH_SHORT).show();
            }
        });
        buttonMultiplicacion.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick (View v){
                int x = Integer.parseInt(editTextA.getText().toString());
                int y = Integer.parseInt(editTextB.getText().toString());
                Operaciones operaciones = new Operaciones(x, y);
                Toast.makeText(CalculadoraActivity.this, "La multiplicacion es :"+operaciones.multiplicacion(), Toast.LENGTH_SHORT).show();
            }
        });

        buttonDivision.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick (View v){
                int x = Integer.parseInt(editTextA.getText().toString());
                int y = Integer.parseInt(editTextB.getText().toString());
                Operaciones operaciones = new Operaciones(x, y);
                Toast.makeText(CalculadoraActivity.this, "La Division es :"+operaciones.division(), Toast.LENGTH_SHORT).show();
            }
        });

        button1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick (View v){
                String valorActual = editTextC.getText().toString();
                valorActual = valorActual + "1";
                editTextC.setText(valorActual);
            }
        });
        button2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick (View v){
                String valorActual = editTextC.getText().toString();
                valorActual = valorActual + "2";
                editTextC.setText(valorActual);
            }
        });
        button3.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick (View v){
                String valorActual = editTextC.getText().toString();
                valorActual = valorActual + "3";
                editTextC.setText(valorActual);
            }
        });
        button4.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick (View v){
                String valorActual = editTextC.getText().toString();
                valorActual = valorActual + "4";
                editTextC.setText(valorActual);
            }
        });
        button5.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick (View v){
                String valorActual = editTextC.getText().toString();
                valorActual = valorActual + "5";
                editTextC.setText(valorActual);
            }
        });
        button6.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick (View v){
                String valorActual = editTextC.getText().toString();
                valorActual = valorActual + "6";
                editTextC.setText(valorActual);
            }
        });
        button7.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick (View v){
                String valorActual = editTextC.getText().toString();
                valorActual = valorActual + "7";
                editTextC.setText(valorActual);
            }
        });
        button8.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick (View v){
                String valorActual = editTextC.getText().toString();
                valorActual = valorActual + "8";
                editTextC.setText(valorActual);
            }
        });
        button9.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick (View v){
                String valorActual = editTextC.getText().toString();
                valorActual = valorActual + "9";
                editTextC.setText(valorActual);
            }
        });
        button0.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick (View v){
                String valorActual = editTextC.getText().toString();
                valorActual = valorActual + "0";
                editTextC.setText(valorActual);
            }
        });
        buttonS.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick (View v){
                String valorActual = editTextC.getText().toString();
                valorActual = valorActual + "+";
                editTextC.setText(valorActual);
            }
        });
        buttonR.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick (View v){
                String valorActual = editTextC.getText().toString();
                valorActual = valorActual + "-";
                editTextC.setText(valorActual);
            }
        });
        buttonM.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick (View v){
                String valorActual = editTextC.getText().toString();
                valorActual = valorActual + "*";
                editTextC.setText(valorActual);
            }
        });
        buttonD.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick (View v){
                String valorActual = editTextC.getText().toString();
                valorActual = valorActual + "/";
                editTextC.setText(valorActual);
            }
        });

        buttonC.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick (View v){
                String valorActual = editTextC.getText().toString();
                valorActual = "";
                editTextC.setText(valorActual);
            }
        });

        Operaciones operaciones = new Operaciones(0, 0);

        buttonIgual.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                try {
                    // Obtén el contenido del EditText
                    String expresion = editTextC.getText().toString();

                    // Divide la expresión en números y operador
                    String[] partes = expresion.split("[+\\-*/]");
                    int numero1 = Integer.parseInt(partes[0]);
                    int numero2 = Integer.parseInt(partes[1]);

                    // Obtiene el operador
                    String operador = expresion.replaceAll("[0-9]", "");

                    // Establece los valores en la instancia de Operaciones
                    operaciones.setA(numero1);
                    operaciones.setB(numero2);

                    // Realiza la operación y muestra el resultado en el EditText
                    int resultado = operaciones.realizarOperacion(operador);
                    editTextC.setText(String.valueOf(resultado));
                } catch (Exception e) {
                    // Maneja posibles errores en la expresión matemática
                    editTextC.setText("Error");
                }
            }
        });
        buttonSalirCalculadora.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                finish();
            }
        });
    }
}

File: PrimerKotlin/app/src/main/java/com/micky/primerkotlin/CalculadoraActivity.kt
package com.micky.primerkotlin

import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.micky.primerkotlin.backend.Operaciones

class CalculadoraActivity : AppCompatActivity() {

    private lateinit var editTextA: EditText
    private lateinit var button1: Button
    private lateinit var button2: Button
    private lateinit var button3: Button
    private lateinit var button4: Button
    private lateinit var button5: Button
    private lateinit var button6: Button
    private lateinit var button7: Button
    private lateinit var button8: Button
    private lateinit var button9: Button
    private lateinit var button0: Button
    private lateinit var buttonS: Button
    private lateinit var buttonR: Button
    private lateinit var buttonM: Button
    private lateinit var buttonD: Button
    private lateinit var buttonC: Button
    private lateinit var buttonSalirCalculadora: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_calculadora)

        editTextA = findViewById(R.id.editTextA)
        button1 = findViewById(R.id.button1)
        button2 = findViewById(R.id.button2)
        button3 = findViewById(R.id.button3)
        button4 = findViewById(R.id.button4)
        button5 = findViewById(R.id.button5)
        button6 = findViewById(R.id.button6)
        button7 = findViewById(R.id.button7)
        button8 = findViewById(R.id.button8)
        button9 = findViewById(R.id.button9)
        button0 = findViewById(R.id.button0)
        buttonS = findViewById(R.id.buttonS)
        buttonR = findViewById(R.id.buttonR)
        buttonM = findViewById(R.id.buttonM)
        buttonD = findViewById(R.id.buttonD)
        buttonC = findViewById(R.id.buttonC)
        buttonSalirCalculadora = findViewById(R.id.buttonSalirCalculadora)

        val buttonIgual: Button = findViewById(R.id.buttonIgual)

        button1.setOnClickListener {
            val valorActual = editTextA.text.toString() + "1"
            editTextA.setText(valorActual)
        }

        button2.setOnClickListener {
            val valorActual = editTextA.text.toString() + "2"
            editTextA.setText(valorActual)
        }

        button3.setOnClickListener {
            val valorActual = editTextA.text.toString() + "3"
            editTextA.setText(valorActual)
        }

        button4.setOnClickListener {
            val valorActual = editTextA.text.toString() + "4"
            editTextA.setText(valorActual)
        }

        button5.setOnClickListener {
            val valorActual = editTextA.text.toString() + "5"
            editTextA.setText(valorActual)
        }

        button6.setOnClickListener {
            val valorActual = editTextA.text.toString() + "6"
            editTextA.setText(valorActual)
        }

        button7.setOnClickListener {
            val valorActual = editTextA.text.toString() + "7"
            editTextA.setText(valorActual)
        }

        button8.setOnClickListener {
            val valorActual = editTextA.text.toString() + "8"
            editTextA.setText(valorActual)
        }

        button9.setOnClickListener {
            val valorActual = editTextA.text.toString() + "9"
            editTextA.setText(valorActual)
        }

        button0.setOnClickListener {
            val valorActual = editTextA.text.toString() + "0"
            editTextA.setText(valorActual)
        }

        buttonS.setOnClickListener {
            val valorActual = editTextA.text.toString() + "+"
            editTextA.setText(valorActual)
        }

        buttonR.setOnClickListener {
            val valorActual = editTextA.text.toString() + "-"
            editTextA.setText(valorActual)
        }

        buttonM.setOnClickListener {
            val valorActual = editTextA.text.toString() + "*"
            editTextA.setText(valorActual)
        }

        buttonD.setOnClickListener {
            val valorActual = editTextA.text.toString() + "/"
            editTextA.setText(valorActual)
        }

        buttonC.setOnClickListener {
            editTextA.setText("")
        }

        val operaciones = Operaciones(0, 0)

        buttonIgual.setOnClickListener {
            try {
                val expresion = editTextA.text.toString()

                val partes = expresion.split("[+\\-*/]".toRegex())
                val numero1 = partes[0].toInt()
                val numero2 = partes[1].toInt()

                val operador = expresion.replace("[0-9]".toRegex(), "")

                operaciones.setA(numero1)
                operaciones.setB(numero2)

                val resultado = operaciones.realizarOperacion(operador)
                editTextA.setText(resultado.toString())
            } catch (e: Exception) {
                editTextA.setText("Error")
            }
        }

        buttonSalirCalculadora.setOnClickListener {
            finish()
        }
    }
}
